### PR TITLE
chore: CPP CI Unit Test Coverage Report

### PR DESCRIPTION
**Pull request by AdityaKasar**

### CODE REVIEW

File: .github/workflows/cxx-build.yml [hunk 1]
```diff
@@ -11,23 +11,31 @@ on:
...
+
 env:
   HUSKY: 0
 jobs:
   thunder:
     name: Build Thunder Libraries
     runs-on: ubuntu-latest
     container:
-      image: kevinshahfws/node-c:3.0
+      image: node:latest
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:25]: Can we manually specify whatever the latest version is here? It wouldn't be the first time that "latest" changed to something that broke a pipeline.

File: .github/workflows/cxx-build.yml [hunk 2]
```diff
@@ -36,9 +44,12 @@ jobs:
 
   openrpc:
     name: Build Openrpc Artifacts
+    needs: thunder
     runs-on: ubuntu-latest
     container:
-      image: kevinshahfws/node-c:3.0
+      image: node:latest
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:50]: Can we manually specify whatever the latest version is here? It wouldn't be the first time that "latest" changed to something that broke a pipeline.

File: .github/workflows/cxx-build.yml [hunk 3]
```diff
@@ -79,24 +90,12 @@ jobs:
...
-          name: discovery
-          path: src/sdks/discovery        
-
   core_sdk:
     name: Build Core SDK
     needs: [thunder, openrpc]
     runs-on: ubuntu-latest
     container:
-      image: kevinshahfws/node-c:3.0
+      image: node:latest
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:98]: Can we manually specify whatever the latest version is here? It wouldn't be the first time that "latest" changed to something that broke a pipeline.

File: .github/workflows/cxx-build.yml [hunk 4]
```diff
@@ -121,27 +119,58 @@ jobs:
...
 
+      - name: Test and Generate coverage report
+        run: .github/workflows/utils.sh generate_core_sdk_coverage_report
+
+      - name: Display coverage report
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: coverage/coverage.cobertura.xml
+          badge: true
+          fail_below_min: true
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:148]: For now let's not fail. I want to eventually but not right now.

File: .github/workflows/cxx-build.yml [hunk 5]
```diff
@@ -121,27 +119,58 @@ jobs:
...
+          path: code-coverage-results.md
+
+
   manage_sdk:
     name: Build Manage SDK
     needs: [thunder, openrpc]
     runs-on: ubuntu-latest
     container:
-      image: kevinshahfws/node-c:3.0
+      image: node:latest
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:168]: Can we manually specify whatever the latest version is here? It wouldn't be the first time that "latest" changed to something that broke a pipeline.

File: .github/workflows/cxx-build.yml [hunk 6]
```diff
@@ -160,22 +189,52 @@ jobs:
...
-  dicovery_sdk:
+      - name: Test and Generate coverage report
+        run: .github/workflows/utils.sh generate_manage_sdk_coverage_report
+
+      - name: Display coverage report
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: coverage/coverage.cobertura.xml
+          badge: true
+          fail_below_min: true
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:218]: See above RE: Not failing

File: .github/workflows/cxx-build.yml [hunk 7]
```diff
@@ -160,22 +189,52 @@ jobs:
...
+          recreate: true
+          path: code-coverage-results.md
+
+  discovery_sdk:
     name: Build Discovery SDK
     needs: [thunder, openrpc]
     runs-on: ubuntu-latest
     container:
-      image: kevinshahfws/node-c:3.0
+      image: node:latest
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:237]: See above RE: image version

File: .github/workflows/cxx-build.yml [hunk 8]
```diff
@@ -199,12 +258,43 @@ jobs:
...
+
+      - name: Test and Generate coverage report
+        run: .github/workflows/utils.sh generate_discovery_sdk_coverage_report
+
+      - name: Display coverage report
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: coverage/coverage.cobertura.xml
+          badge: true
+          fail_below_min: true
```
Comment (kschrief) [inline .github/workflows/cxx-build.yml:287]: See above RE: not failing

**Review by kschrief — CHANGES_REQUESTED** [review]

File: .github/workflows/cxx-build.yml (hunk 5 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml]: fixed

File: .github/workflows/cxx-build.yml (hunk 4 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml:160]: even if ctest fail. Git action steps won't be interrupted for now.

File: .github/workflows/cxx-build.yml (hunk 3 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml]: fixed

File: .github/workflows/cxx-build.yml (hunk 6 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml:230]: even if ctest fail. Git action steps won't be interrupted for now.

File: .github/workflows/cxx-build.yml (hunk 7 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml]: fixed

File: .github/workflows/cxx-build.yml (hunk 8 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml:299]: even if ctest fail. Git action steps won't be interrupted for now.

File: .github/workflows/cxx-build.yml (hunk 2 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml]: fixed

File: .github/workflows/cxx-build.yml (hunk 1 above)
Comment (AdityaKasar) [reply .github/workflows/cxx-build.yml]: fixed

**Pull request merged**